src/nvim/menu.c

```diff
@@ -878,8 +878,9 @@ char_u *set_context_in_menu_cmd(expand_T *xp, char_u *cmd, char_u *arg, int forc
 
     menu = root_menu;
     if (after_dot != arg) {
-      path_name = xmalloc(after_dot - arg);
-      STRLCPY(path_name, arg, after_dot - arg);
+      size_t path_len = (size_t) (after_dot - arg);
+      path_name = xmalloc(path_len);
+      STRLCPY(path_name, arg, path_len);
```

Comment:
If I'm not mistaken, `(after_dot != arg)` should be changed to `(after_dot > arg)` (line 880). The logic is the same (or stronger), and it guarantees (size_t) cast is valid.